osu.Framework.Android/AndroidGameActivity.cs

```diff
@@ -90,6 +93,32 @@ protected override void OnCreate(Bundle savedInstanceState)
                             Window.ClearFlags(WindowManagerFlags.KeepScreenOn);
                     });
                 }, true);
+
+                host.ScreenOrientation.BindValueChanged(e =>
+                {
+                    // Don't do anything if orientation is locked
+                    if (host.LockScreenOrientation.Value) return;
+
+                    RunOnUiThread(() =>
+                    {
+                        RequestedOrientation = configToNativeOrientationEnum(e.NewValue);
+                    });
+                });
+                host.LockScreenOrientation.BindValueChanged(e =>
+                {
+                    // Throw if consumer change locked state while ScreenOrientation bindable is disabled
+                    if (host.ScreenOrientation.Disabled)
+                        throw new InvalidOperationException("Can't change screen orientation lock when setting is disabled");
+
+                    if (e.NewValue) // If locked
+                    {
+                        RunOnUiThread(() =>
+                        {
+                            RequestedOrientation = NativeOrientation.Locked;
+                        });
+                    }
+                    else host.ScreenOrientation.TriggerChange();
```

Comment:
This is really bad. Split out the required logic to its own method rather than calling `TriggerChange`.

Or even better, simplify this whole block to:

```suggestion
  RunOnUiThread(() =>
                         {
                             RequestedOrientation = e.NewValue ? NativeOrientation.Locked : configToNativeOrientationEnum(host.ScreenOrientation.Value);
                         });
                    else host.ScreenOrientation.TriggerChange();
```

Reply:
Changed in next commits.